src/sentry/tasks/performance_detection.py

```diff
@@ -0,0 +1,303 @@
+import hashlib
+import random
+from abc import ABC, abstractmethod
+from datetime import timedelta
+from enum import Enum
+from typing import Any, Dict, List
+
+import sentry_sdk
+
+from sentry import options
+from sentry.eventstore.processing.base import Event
+
+Span = Dict[str, Any]
+TransactionSpans = List[Span]
+PerformanceIssues = Dict[str, Any]
+
+
+class DetectorType(Enum):
+    SLOW_SPAN = "slow_span"
+    DUPLICATE_SPANS = "duplicate_spans"
+    SEQUENTIAL_SLOW_SPANS = "sequential_slow_spans"
+
+
+# Facade in front of performance detection to limit impact of detection on our events ingestion
+def detect_performance_issue(data: Event):
+    try:
+        rate = options.get("store.use-ingest-performance-detection-only")
+        if rate and rate > random.random():
+            # Add an experimental tag to be able to find these spans in production while developing. Should be removed later.
+            sentry_sdk.set_tag("_did_analyze_performance_issue", "true")
+            with sentry_sdk.start_span(
+                op="py.detect_performance_issue", description="none"
+            ) as sdk_span:
+                _detect_performance_issue(data, sdk_span)
```

Comment:
I would spawn those tasks onto a separate queue as a followup. In the meantime I would move this code out of `tasks`... at first glance i thought you created a celery task here because of that.